### PR TITLE
Feat: API pagination for the streams endpoint

### PR DESCRIPTION
**Pull request by ya7ya**
**What does this pull request do? Explain your changes. (required)**
adds `limit` and `offset` capabilities to the API so it won't list the entire list of streams every time. and it can be used for pagination

**Specific updates (required)**
- updates to `leveldb` to use `limit` 
- updates to the `postgresql` to use `limit` and `offset`
- 

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional):**

**Checklist:**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/store/level-store.js [hunk 1]
```diff
@@ -43,11 +48,16 @@ export default class LevelStore {
     }
   }
 
-  async list(prefix = '') {
+  async list(prefix = '', limit = DEFAULT_LIMIT, offset = 0) {
```
Comment (iameli) [inline packages/api/src/store/level-store.js:51]: This is fine for now; it's hard for me to imagine a LevelDB API user with thousands of thousands of streams. We will eventually need a more sophisticated way to handle query continuations but I can file that as a separate enhancement.

File: packages/api/package.json [hunk 2]
```diff
@@ -36,6 +36,7 @@
     "cross-env": "^5.2.0",
     "express": "^4.16.4",
     "express-async-errors": "^3.1.1",
+    "express-bearer-token": "^2.4.0",
```
Comment (iameli) [inline packages/api/package.json]: Mind doing a `git rebase -i eli/api-node-prototype` and taking out the auth commits so this can merge separately?

**Review by iameli — CHANGES_REQUESTED** [review]
Looks solid if you can rebase out the auth commits. I'll file a separate issue for a "continuation token" kind of design.

File: packages/api/src/store/level-store.js (hunk 1 above)
Comment (ya7ya) [reply packages/api/src/store/level-store.js:51]: yeah, I'll figure out a more efficient way then 👍

**Review by iameli — APPROVED** [review]

**Comment by iameli** [comment]
LGTM! I'll merge to bypass the failing CI, hopefully fixing that today.

**Pull request merged**